### ISSUE TITLE
More robust check for regeneration [WIP]

### ISSUE THREAD

**Issue by ZanSara** [addisonElliott/pyqt5ac#11]
Currently, pyqt5ac verifies whether a file needs to be regenerated only basing on the last saved date recorded on the file system.

However, a generated file might become invalid for multiple other reasons:
- IDE's refactoring (or not refactoring properly) imports in the generated files and not in the *.ui / *.qrc
- Other files, which are being imported into the current one, changing name/path (like *.qrc's resources)
- Users editing the files
- Probably more unexpected ones

I would recommend to plan a strategy to detect such changes and how to deal with them.

**Comment by zkovari** [comment]
I don't think we should detect when the generated files are changed. For me it doesn't seem this module's responsibility.

What we further check is if the included files in a qrc resource are outdated or not. For this one I also added a test case: https://github.com/addisonElliott/pyqt5ac/blob/master/tests/test_pyqt5ac.py#L172

But if the path of any included file in a resource would change meantime, that's I'm not if we detect  or not. I will add a test case for that when I have some time. Did you check already what happens in that case?

**Comment by ZanSara** [comment]
Maybe it's currently not it's responsibility, but it could be a welcome addition. After all, it happens rather often that I run on generated files believing everything is ok (because the IDE silently fixed the imports) and then, when long after I edit and recompile their .ui, everything breaks. Having the CI recompile the .ui every time is a big help, but preventing would be better.

Regarding what happens if I move files, so far I didn't test it too much yet, I'll wait for the custom vars and the `__init__.py` check before promoting it to anyone but myself. However, from the code:
```
outdated = (not os.path.exists(dst) or (os.path.getmtime(src) > os.path.getmtime(dst)))
```
It seems like only the last editing time is relevant. It might be even OS dependend what does this value returns. The doc doesn't specify:

>  os.path.getmtime(path)
> Return the time of last modification of 'path'. The return value is a number giving the number of seconds since the epoch (see the time module). Raise OSError if the file does not exist or is inaccessible.
https://docs.python.org/3.6/library/os.path.html#os.path.getmtime

Probably moving files or renaming a folder along the path does not correspond to an edit, and therefore imports could break.
